File: Source Code/Samples/Bölüm 11/Senaryo/CS/Senaryo/Model/OperasyonListesi.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

using System.Text;

namespace Model
{
    public sealed class OperasyonListesi: KeyedCollection<string, Operasyon>
    {
        private static OperasyonListesi liste;

        public static OperasyonListesi Liste
        {
            get
            {
                return liste;
            }
        }

        static OperasyonListesi()
        {
            liste = new OperasyonListesi();
            liste.Add(new Operasyon("MüþteriEkle", "Müþteri Ekleme", "Ýþ", "Ýþ.MüþteriÝþSýnýfý", "Ekle"));
            liste.Add(new Operasyon("MüþteriAra", "Müþteri Arama", "Ýþ", "Ýþ.MüþteriÝþSýnýfý", "Ara"));
            liste.Add(new Operasyon("MüþteriSil", "Müþteri Silme", "Ýþ", "Ýþ.MüþteriÝþSýnýfý", "Sil"));
            liste.Add(new Operasyon("MüþteriListele", "Müþteri Arama", "Ýþ", "Ýþ.MüþteriÝþSýnýfý", "Listele"));

        }

        protected override string GetKeyForItem(Operasyon item)
        {
            return item.Kod;
        }

        private OperasyonListesi()
        {
        }


    }
}

File: Source Code/Samples/Bölüm 11/Senaryo/VB/Senaryo/Model/OperasyonListesi.vb
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.Text


Public NotInheritable Class OperasyonListesi
    Inherits KeyedCollection(Of String, Operasyon)


    Private Shared _liste As OperasyonListesi

    Shared Sub New()
        _liste = New OperasyonListesi()
        _liste.Add(New Operasyon("MüþteriEkle", "Müþteri Ekleme", "Ýþ", "Ýþ.MüþteriÝþSýnýfý", "Ekle"))
        _liste.Add(New Operasyon("MüþteriAra", "Müþteri Arama", "Ýþ", "Ýþ.MüþteriÝþSýnýfý", "Ara"))
        _liste.Add(New Operasyon("MüþteriSil", "Müþteri Silme", "Ýþ", "Ýþ.MüþteriÝþSýnýfý", "Sil"))
        _liste.Add(New Operasyon("MüþteriListele", "Müþteri Arama", "Ýþ", "Ýþ.MüþteriÝþSýnýfý", "Listele"))
    End Sub

    Private Sub New()

    End Sub

    Public Shared ReadOnly Property Liste() As OperasyonListesi
        Get
            Return _liste
        End Get
    End Property

    Protected Overrides Function GetKeyForItem(ByVal item As Operasyon) As String
        Return item.Kod
    End Function
End Class
